lightning/src/routing/router.rs

```diff
@@ -429,23 +451,54 @@ pub struct PaymentParameters {
 	/// these SCIDs.
 	pub previously_failed_channels: Vec<u64>,
 
-	/// The minimum CLTV delta at the end of the route.
-	///
-	/// This field should always be set to `Some` and may be required in a future release.
-	pub final_cltv_expiry_delta: Option<u32>,
+	/// The minimum CLTV delta at the end of the route. This value must not be zero.
+	pub final_cltv_expiry_delta: u32,
+}
+
+impl Writeable for PaymentParameters {
```

Comment:
Why not keep using the macro but default to 0 (or `MIN_CLTV_EXPIRY_DELTA`), so the field can be set by higher-level reads? Do we expect users to be serializing this struct?

Reply:
Hmm, if we kept `PayParams` using the ser macro with a default value, `RouteParams` deser would always ensure it sets the correct value, though. I see why we need to break out of the macro for `RouteParams`, but just not sure I understand why we need to break out for `PaymentParams`. 

> If a user has a RouteParameters serialized in 113, eg because of a payment failure, and they retry it on restart in 114

\+ retries on restart aren't supported in 114 (do you mean using `send_payment`?).

Reply:
We wouldn't be synthesizing the wrong value, though, IIUC. Shouldn't this code in `RouteParams` deser cover this case, if they do want to use `PaymentPathFailed::retry`?
```
		let mut payment_params: PaymentParameters = payment_params.0.unwrap();
		if payment_params.final_cltv_expiry_delta == 0 {
			payment_params.final_cltv_expiry_delta = final_cltv_expiry_delta.0.unwrap();
		}
```
Another thought could be removing `retry` entirely, which we want to do at some point anyway, maybe even more ser complexity introduced there though.

Reply:
Anyway, we may be talking past each other, and it doesn't matter that much though the broken-up macro is somewhat ugly, so I won't die on this hill :)

Reply:
Yeah, just seems unlikely this struct would ever be serialized on its own, even if it's technically possible. Fine to leave as-is